cli/src/semgrep/project.py

```diff
@@ -24,3 +38,71 @@ def get_project_url() -> Optional[str]:
         except Exception as e:
             logger.debug(f"Failed to get project url from .git/config: {e}")
             return None
+
+
+@define
+class ProjectConfig:
+    metadata: Dict[str, Any] = Factory(dict)
+
+    @staticmethod
+    def is_project_config_file(file_path: Path) -> bool:
+        return CONFIG_FILE_PATTERN.search(file_path.name) is not None
+
+    @classmethod
+    def _find_all_config_files(cls, src_directory: Path, cwd_path: Path) -> List[Path]:
+        conf_files = []
+
+        # Populate stack of directories to traverse
+        stack = {cwd_path}
+        temp_path = src_directory
+        dir_route = cwd_path.relative_to(src_directory)
+        for parent in dir_route.parents:
+            temp_path = temp_path / parent
+            stack.add(temp_path)
+
+        # Traverse stack looking for config files
+        while stack:
+            cur_path = stack.pop()
+            conf_files += [
+                f for f in cur_path.iterdir() if cls.is_project_config_file(f)
+            ]
+        return conf_files
+
+    @classmethod
+    def load_from_file(cls, file_path: Path) -> Dict[str, Any]:
+        yaml = ruamel.yaml.YAML(typ="safe")
+        logger.debug(f"Loading semgrepconfig file: {file_path}")
+        with file_path.open("r") as fp:
+            config: Dict[str, Any] = yaml.load(fp)
+            return config
+
+    @classmethod
+    def load_all(cls) -> "ProjectConfig":
+        src_directory = get_git_root_path()
+        cwd_path = Path.cwd()
+        conf_files = cls._find_all_config_files(src_directory, cwd_path)
+
+        # Sort by depth asc
```

Comment:
This means deeper files are later in list and thus take precedence, yes?

Reply:
That is correct!